src/sentry/plugins/sentry_webhooks/plugin.py

```diff
@@ -118,17 +119,19 @@ def get_group_data(self, group, event, triggering_rules):
     def get_webhook_urls(self, project):
         return split_urls(self.get_option("urls", project))
 
-    def send_webhook(self, url, payload):
-        return safe_urlopen(url=url, json=payload, timeout=self.timeout, verify_ssl=False)
+    def get_client(self, payload):
+        return WebhookApiClient(payload)
 
     def notify_users(self, group, event, triggering_rules, fail_silently=False, **kwargs):
         payload = self.get_group_data(group, event, triggering_rules)
+        client = self.get_client(payload)
         for url in self.get_webhook_urls(group.project):
-            # TODO: Use API client with raise_error
-            safe_execute(
-                self.send_webhook,
-                url,
-                payload,
-                _with_transaction=False,
-                expected_errors=(ReadTimeout, ConnectionError),
-            )
+            try:
```

Comment:
I think in python you can do 
```py
try:
    client.request(url)
except (ReadTimeout, ConnectionError):
    pass
```
and it will re-raise the Exception in all other cases.